diff -Nur vodcatcher-0.1.1.orig/po/vodcatcher.pot vodcatcher-0.1.1/po/vodcatcher.pot
--- vodcatcher-0.1.1.orig/po/vodcatcher.pot	2007-11-27 19:41:15.000000000 +0100
+++ vodcatcher-0.1.1/po/vodcatcher.pot	2008-07-07 23:04:05.000000000 +0200
@@ -8,7 +8,7 @@
 msgstr ""
 "Project-Id-Version: PACKAGE VERSION\n"
 "Report-Msgid-Bugs-To: <tg@e-tobi.net>\n"
-"POT-Creation-Date: 2007-11-27 19:41+0100\n"
+"POT-Creation-Date: 2008-07-07 23:04+0200\n"
 "PO-Revision-Date: YEAR-MO-DA HO:MI+ZONE\n"
 "Last-Translator: FULL NAME <EMAIL@ADDRESS>\n"
 "Language-Team: LANGUAGE <LL@li.org>\n"
diff -Nur vodcatcher-0.1.1.orig/src/ItemMenuPresenter.cc vodcatcher-0.1.1/src/ItemMenuPresenter.cc
--- vodcatcher-0.1.1.orig/src/ItemMenuPresenter.cc	2007-11-27 18:49:53.000000000 +0100
+++ vodcatcher-0.1.1/src/ItemMenuPresenter.cc	2008-07-08 15:42:10.000000000 +0200
@@ -20,6 +20,7 @@
  *
  */
 
+#include <curl/curl.h>
 #include <vdr/player.h>
 #include "ItemMenuPresenter.h"
 #include "IServiceLocator.h"
@@ -80,6 +81,7 @@
 
 eOSState ItemMenuPresenter::ProcessItemAction(eKeys key, Item item)
 {
+    ostringstream os;
     switch (key)
     {
         case kOk:
@@ -118,6 +120,19 @@
             }
             UpdateYellowButton();
             break;
+        case kBlue:
+			CURL *curl;
+			curl = curl_easy_init();
+	
+			os << "http://";
+			os << "localhost";
+			os << ":";
+			os << "8080";
+			os << "/downloads?action=download_item&url=";
+			os << curl_easy_escape(curl, item.GetVideoCastStream(_configuration.GetPlayBackQuality()).c_str(), 0);
+			serviceLocator.GetDownloadPool()->AddDownload(os.str());
+			curl_easy_cleanup(curl);
+        	break;
         default:
             break;
     }
@@ -225,6 +240,7 @@
         {
             menu->SetGreenHelp(tr("Play"));
             menu->SetRedHelp(tr("Record"));
+            menu->SetBlueHelp("Download");
             UpdateYellowButton();
         }
     }
